File: Selenium/Activities/Activity5.py

from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("https://www.v1.training-support.net/selenium/input-events")
    service = FirefoxService(GeckoDriverManager().install())
    print("Page title is: ", driver.title)

    # Perform a LEFT CLICK action
    actions = ActionChains(driver)
    actions.click().perform()
    # Find the side that is active
    active_side = driver.find_element(By.CLASS_NAME, "active")
    # Print the number shown
    print("Active side number after left click: ", active_side.text)
    
    # Perform a DOUBLE CLICK action
    actions.double_click().perform()
    # Find the side that is active
    active_side = driver.find_element(By.CLASS_NAME, "active")
    # Print the number shown
    print("Active side number after double click: ", active_side.text)

    # Perform a RIGHT CLICK action
    actions.context_click().perform()
    # Find the side that is active
    active_side = driver.find_element(By.CLASS_NAME, "active")
    # Print the number shown
    print("Active side number after right click: ", active_side.text)
    driver.quit()

File: Selenium/Activities/Activity5.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity5 {

    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();

        driver.get("https://v1.training-support.net/selenium/input-events");
        System.out.println("Title is " +driver.getTitle());

        Actions builder = new Actions(driver);
        builder.click().pause(1000).build().perform();
        String frontTextLeftclick = driver.findElement(By.className("active")).getText();
         System.out.println("Front text on left click is " +frontTextLeftclick);

        builder.doubleClick().pause(1000).build().perform();
        String frontTextDoubleclick = driver.findElement(By.className("active")).getText();
        System.out.println("Number on Double click is " +frontTextDoubleclick);

        builder.contextClick().pause(1000).build().perform();
        String frontTextrightclick = driver.findElement(By.className("active")).getText();
        System.out.println("Front text on right click is " +frontTextrightclick);

         driver.quit();


    }
}
